astropy/utils/metadata/merge.py

```diff
@@ -115,6 +81,33 @@ class method.  This takes ``left`` and ``right`` side arguments and
 
     # types = [(left_types, right_types), ...]
 
+    def __init_subclass__(cls):
+        members = vars(cls)
+
+        # Wrap ``merge`` classmethod to catch any exception and re-raise as
+        # MergeConflictError.
+        if "merge" in members and isinstance(members["merge"], classmethod):
```

Comment:
I'd tend to rewrite this a bit to make use of the fact that `cls.merge` may exist:
```
if isinstance((merge := getattr(cls, "merge", None)), classmethod):
```
and then propagate it below.

Reply:
✅

Reply:
That doesn't appear to work.

Reply:
it does when using `members.get()`